### PR TITLE
mesa: 18.0.3 -> 18.1.2

### PR DESCRIPTION
**Pull request by mkaito**
###### Motivation for this change
Update to latest stable mesa

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by mkaito** [comment]
I've got another PR coming up for mesa that fixes pkgconfig paths for some things. Maybe merge them together to avoid doing the massive rebuild twice.

**Comment by xeji** [comment]
Please rebase this PR on the `staging` branch since it causes a mass rebuild.

**Comment by mkaito** [comment]
rebased

File: pkgs/development/libraries/mesa/default.nix [hunk 1]
```diff
@@ -2,7 +2,7 @@
 , pkgconfig, intltool, autoreconfHook, substituteAll
 , file, expat, libdrm, xorg, wayland, wayland-protocols, openssl
 , llvmPackages, libffi, libomxil-bellagio, libva-minimal
-, libelf, libvdpau, valgrind-light, python2
+, libelf, libvdpau, valgrind-light, python2, python27Packages
```
Comment (FRidh) [inline pkgs/development/libraries/mesa/default.nix]: pass in either `python` or `pythonPackages`, not both.

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/mesa/default.nix (hunk 1 above)
Comment (mkaito) [reply pkgs/development/libraries/mesa/default.nix]: Can I reference `Mako` (why is that capitalized?) from `python2` instead of `python27Packages`?

Comment (xeji) [reply pkgs/development/libraries/mesa/default.nix]: You can reference packages like this:  `python2.pkgs.Mako`.

**Comment by mkaito** [comment]
Done.

I've been running it for a few days now. Seems stable, and actually seems to improve performance on some games.

**Comment by Chiiruno** [comment]
>I've been running it for a few days now. Seems stable, and actually seems to improve performance on some games.
Mesa implements Vulkan right? Hopefully it'll fix that EFB problem I've been having with `dolphin-emu`.

**Pull request merged**